tests/suites/test_suite_psa_crypto_slot_management.function

```diff
@@ -639,23 +651,50 @@ void invalid_handle( )
     psa_set_key_algorithm( &attributes, 0 );
     PSA_ASSERT( psa_import_key( &attributes,
                                 material, sizeof( material ),
-                                &handle1 ) );
-    TEST_ASSERT( handle1 != 0 );
+                                &valid_handle ) );
+    TEST_ASSERT( valid_handle != 0 );
 
-    /* Attempt to close and destroy some invalid handles. */
-    TEST_EQUAL( psa_close_key( 0 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_close_key( handle1 - 1 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_close_key( handle1 + 1 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_destroy_key( 0 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_destroy_key( handle1 - 1 ), PSA_ERROR_INVALID_HANDLE );
-    TEST_EQUAL( psa_destroy_key( handle1 + 1 ), PSA_ERROR_INVALID_HANDLE );
+    /* Construct an invalid handle as specified in the test case data. */
+    switch( handle_construction )
+    {
+        case INVALID_HANDLE_0:
+            invalid_handle = 0;
+            break;
+        case INVALID_HANDLE_UNOPENED:
+            /* We can't easily construct a handle that's never been opened
+             * without knowing how the implementation constructs handle
+             * values. The current test code assumes that valid handles
+             * are in a range between 1 and some maximum. */
+            if( valid_handle == 1 )
+                invalid_handle = 2;
+            else
+                invalid_handle = valid_handle - 1;
```

Comment:
Do we sometimes want to test `invalid_handle = valid_handle + 1`? We may need a `INVALID_HANDLE_UNOPENED_ABOVE` for this, renaming the `- 1` case to `INVALID_HANDLE_UNOPENED_BELOW`.

Reply:
I think it would be better to test both, but you objected to that.

Reply:
I objected to the tests silently only testing one way without letting us know we reduced test coverage.

Reply:
So it's a misunderstanding then? The tests were testing both ways, skipping one of the ways if it was not applicable.

Reply:
I didn't like skipping if one wasn't applicable, as both might be silently skipped. I'd like to always see both ways tested (above and below). If this means we have to allocate multiple keys, we should do that.

Reply:
No, it was not possible for both to be silently skipped. `handle1 - 1 != 0` and `handle1 + 1 != 0` can't both be false.

Reply:
OK